axelor-production/src/main/resources/views/ManufOrder.xml

```diff
@@ -233,6 +233,16 @@
     	<attribute for="isConsProOnOperation" name="readonly" expr="eval: !(prodProcess?.isConsProOnOperation)"/>
     </action-attrs>
     
+    <action-attrs name="action-manuf-order-atts-sale-order-onchange-client-partner">
```

Comment:
Domain action-attrs must be applied on onSelect not on onChange. To reset saleOrder you can create another action and that action should only apply if select partner is different then clientPartner from saleOrder.